### ISSUE TITLE
[Question] How can I search by exact match?

### ISSUE THREAD

**Issue by burritoIand** [elastic/elasticsearch-dsl-py#1339]
How can I have a field that I can filter by with an exact match?
```
elasticsearch==7.6.0
elasticsearch-dsl==7.1.0
```

# es model
```
class Post(Document):
    group_id = Text(fields={'term': Keyword()})
```

# query
```
>>> Post.search().query('match', group_id='toast').execute() // is not using the keyword 'term'
<Response: [Post(index='posts-staging', id='TAm48k8fzq5QVQ')]>

>>> Post.search().query('match', group_id__term='toast').execute() // this is the one I'm questioning in particular.
<Response: {}>

>>> Post.search().query('match', group_id='toasta').execute()
<Response: {}>
>>> Post.search().query('match', group_id='1toast').execute()
<Response: {}>
```

Would it be better to instead just use a Keyword as the type?
```
class Post(Document):
    group_id = Keyword()
```
And then just do `.query('match', group_id='toast')`? Is it that I'm just losing the ability for full-text search on that field?

**Comment by cj564932957** [comment]
use term for keyword

**Comment by cj564932957** [comment]
Dotted fields
Sometimes you want to refer to a field within another field, either as a multi-field (title.keyword) or in a structured json document like address.city. To make it easier, the Q shortcut (as well as the query, filter, and exclude methods on Search class) allows you to use __ (double underscore) in place of a dot in a keyword argument:

s = Search()
s = s.filter('term', category__keyword='Python')
s = s.query('match', address__city='prague')
Alternatively you can always fall back to python’s kwarg unpacking if you prefer:

s = Search()
s = s.filter('term', **{'category.keyword': 'Python'})
s = s.query('match', **{'address.city': 'prague'})

**Comment by burritoIand** [comment]
ok great thanks, I ended up changing it to `group_id = Keyword()` and then doing `search = search.filter('term', group_id=group_id)`. After reading the ES docs I realized that for one I should be doing a filter not a query (since this refinement shouldn't affect the "score" of the search results). From my understanding the only downside would be if I wanted to enable full-text search on the group_id field, but I can't see why I would? Anyways thanks for the explanation!
